File: 2642-design-graph-with-shortest-path-calculator/2642-design-graph-with-shortest-path-calculator.java
class Graph {

    int[][] adj;
    int n;
    public Graph(int n, int[][] edges) {
        adj = new int[n][n];
        this.n = n;
        for(int[] a: adj) {
            Arrays.fill(a, (int)1e9);
        }

        for(int[] e: edges) {
            adj[e[0]][e[1]] = e[2];
        }

        for(int i = 0; i < n; i++) {
            adj[i][i] = 0;
        }

        for(int k = 0; k < n; k++) {
            for(int i = 0; i < n; i++) {
                for(int j = 0; j < n; j++) {
                    adj[i][j] = Math.min(adj[i][j], adj[i][k] + adj[k][j]);
                }
            }
        }
    }
    
    public void addEdge(int[] e) {
        
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                adj[i][j] = Math.min(adj[i][j], adj[i][e[0]] + adj[e[1]][j] + e[2]);
            }
        }
    }
    
    public int shortestPath(int node1, int node2) {
        
        if(adj[node1][node2] >= (int)1e9) return -1;
        return adj[node1][node2];
    }
}

/**
 * Your Graph object will be instantiated and called as such:
 * Graph obj = new Graph(n, edges);
 * obj.addEdge(edge);
 * int param_2 = obj.shortestPath(node1,node2);
 */

File: 2642-design-graph-with-shortest-path-calculator/2642-design-graph-with-shortest-path-calculator.js
var Graph = function(n, edges) {
    this.n = n;
    this.distance = Array.from({ length: n }, () => Array(n).fill(Number.MAX_SAFE_INTEGER));

    for (let i = 0; i < n; i++) {
        this.distance[i][i] = 0;
    }

    for (const edge of edges) {
        this.distance[edge[0]][edge[1]] = edge[2];
    }

    for (let k = 0; k < n; k++) {
        for (let i = 0; i < n; i++) {
            for (let j = 0; j < n; j++) {
                this.distance[i][j] = Math.min(
                this.distance[i][j],
                this.distance[i][k] + this.distance[k][j]
                );
            }
        }
    }
};
Graph.prototype.addEdge = function(edge) {
    if (this.distance[edge[0]][edge[1]] <= edge[2]) {
            return;
        }
        this.distance[edge[0]][edge[1]] = edge[2];

        for (let i = 0; i < this.n; i++) {
            for (let j = 0; j < this.n; j++) {
                this.distance[i][j] = Math.min(
                    this.distance[i][j],
                    this.distance[i][edge[0]] + edge[2] + this.distance[edge[1]][j]
                );
            }
        }
};
Graph.prototype.shortestPath = function(node1, node2) {
    if (this.distance[node1][node2] === Number.MAX_SAFE_INTEGER) {
            return -1;
        }
        return this.distance[node1][node2];
};